pkg/commands/testdata/replace-update-strategy/replace-annotations.yaml

```diff
@@ -0,0 +1,28 @@
+# Copyright 2022 VMware, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: carto.run/v1alpha1
+kind: Workload
+metadata:
+  name: spring-petclinic
+  annotations:
```

Comment:
## I'm not sure this is correct.
The workload object passes all annotation references to the supply chain via a param named "annotations"
And the supply chain converts the annotation param values to actual kubernetes annotations on the backend.

### Here's an example of a workload.yaml with some annotations that were added via the apps CLI plugin:

```
ta apply tanzu-java-web-app  --annotation autoscaling.knative.dev/min-scale=2 --annotation autoscaling.knative.dev/max-scale=5
🔎 Update workload:
...
  6,  6   |    apps.tanzu.vmware.com/workload-type: web
  7,  7   |  name: tanzu-java-web-app
  8,  8   |  namespace: ns1
  9,  9   |spec:
     10 + |  params:
     11 + |  - name: annotations
     12 + |    value:
     13 + |      autoscaling.knative.dev/max-scale: "5"
     14 + |      autoscaling.knative.dev/min-scale: "2"
...

ta get tanzu-java-web-app --export
---
apiVersion: carto.run/v1alpha1
kind: Workload
metadata:
  labels:
    apps.tanzu.vmware.com/workload-type: web
  name: tanzu-java-web-app
  namespace: ns1
spec:
  params:
  - name: annotations
    value:
      autoscaling.knative.dev/max-scale: "5"
      autoscaling.knative.dev/min-scale: "2"
  source:
    image: gcr.io/pivotal-knative/jc-apps:latest@sha256:26d0e12a3e6e1a1d74bf18d52f3d39aa5e9555a7b69b419e4684bd36b77aa749
```

### And if I inspect the running pod, you can see those have been transformed into true annotations (here's a snippet of the output):
```
k get pod tanzu-java-web-app-00002-deployment-7fdf4b8fbb-rhrfj
NAME                                                   READY   STATUS    RESTARTS   AGE
tanzu-java-web-app-00002-deployment-7fdf4b8fbb-rhrfj   2/2     Running   0          3m37s
11/16/22-16:22:43 ~ %k describe pod tanzu-java-web-app-00002-deployment-7fdf4b8fbb-rhrfj
Name:             tanzu-java-web-app-00002-deployment-7fdf4b8fbb-rhrfj
Namespace:        ns1
Priority:         0
Service Account:  default
Node:             gke-iterate-1-4-x-default-pool-61cc6a50-1cks/10.150.0.51
Start Time:       Wed, 16 Nov 2022 16:19:06 -0800
Labels:           app=tanzu-java-web-app-00002

                  ...

Annotations:      autoscaling.knative.dev/max-scale: 5
                  autoscaling.knative.dev/min-scale: 2
                 
                  ...
         
Events:
  Type     Reason     Age                    From               Message
  ----     ------     ----                   ----               -------
  Normal   Scheduled  3m48s                  default-scheduler  Successfully assigned ns1/tanzu-java-web-app-00002-deployment-7fdf4b8fbb-rhrfj to gke-iterate-1-4-x-default-pool-61cc6a50-1cks
  
  ...

  Warning  Unhealthy  3m41s (x3 over 3m43s)  kubelet            Readiness probe failed: Get "http://10.4.3.32:8012/": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```

Reply:
so what should happen if the user does something like this? Directly add an annotation into the workload metadata. Should it fail? 

Because in another test (refer to the integration test), these annotations are being passed through flags, and in the workload yaml they're added as params. Both ways are currently working.

Reply:
@shashwathi any thoughts?

Reply:
Conclusion is: both fields are allowed to be set in `yaml` file, and there are test cases covering both scenarios.